dags/ecocounter_pull.py

```diff
@@ -0,0 +1,190 @@
+r"""### Daily Ecocounter Data Pull DAG
+Pipeline to pull ecocounter daily data and insert them into Postgres tables using Python Operator.
+Also runs SQL data-checks on the pulled data, checking row count and distinct flow_ids compared to a lookback period.
+"""
+
+#add logging, unknown ids formatting
+#start date and end date?
+
+import sys
+import os
+import pendulum
+from datetime import timedelta
+import dateutil.parser
+import logging
+from configparser import ConfigParser
+
+from airflow.decorators import dag, task, task_group
+from airflow.models import Variable
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.macros import ds_add
+from airflow.exceptions import AirflowSkipException
+
+try:
+    repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+    sys.path.insert(0, repo_path)
+    from dags.dag_functions import (
+        task_fail_slack_alert, send_slack_msg, get_readme_docmd
+    )
+    from dags.custom_operators import SQLCheckOperatorWithReturnValue
+    from volumes.ecocounter.pull_data_from_api import (
+        getToken, getSites, getFlowData, siteIsKnownToUs, insertSite,
+        insertFlow, flowIsKnownToUs, truncateFlowSince, insertFlowCounts,
+        getKnownSites, getKnownFlows
+    )
+except:
+    raise ImportError("Cannot import DAG helper functions.")
+
+DAG_NAME = 'ecocounter_pull'
+DAG_OWNERS = Variable.get('dag_owners', deserialize_json=True).get(DAG_NAME, ["Unknown"])
+
+README_PATH = os.path.join(repo_path, 'volumes/ecocounter/readme.md')
+DOC_MD = get_readme_docmd(README_PATH, DAG_NAME)
+
+CONFIG_PATH = '/data/airflow/data_scripts/volumes/ecocounter/.api-credentials.config'
```

Comment:
wondering if we should put this as an airflow connection instead 🤔

Reply:
Will do!